src/EntityComponentManager.cc

```diff
@@ -1224,3 +1224,9 @@ std::unordered_set<ComponentTypeId> EntityComponentManager::ComponentTypes(
 
   return result;
 }
+
+/////////////////////////////////////////////////
+void EntityComponentManager::SetEntityCreateOffset(uint64_t _offset)
+{
+  this->dataPtr->entityCount = _offset;
```

Comment:
I can see this being an issue is `_offset < this->dataPtr->entityCount`.

Reply:
Yup, this function could be dangerous if used improperly.

Reply:
Warning added in d9001c7c84e3fb8f9d6a0e19c7472601e1515729